meta.go

```diff
@@ -272,6 +274,36 @@ func (t *TermID) Validate() error {
 	return nil
 }
 
+// String returns the term in it's canonical string format.
+// This would be one of:
+//   tenant:owner/name/revision
+//   owner/name/revision
+//   owner/name
+//   name/revision
+//   name
+func (t *TermsId) String() string {
+	id := ""
+	if t.Tenant != "" {
+		id = fmt.Sprintf("%s:", t.Tenant)
```

Comment:
I find this a little hard to read, and it could be a bit more efficient too.

How about this, which is 5 lines shorter and quite a bit more efficient?

	func (t *TermsId) String() string {
		id := make([]byte, 0, len(t.Tenant)+1+len(t.Owner)+1+len(t.Name)+4)
		if t.Tenant != "" {
			id = append(id, t.Tenant...)
			id = append(id, ':')
		}
		if t.Owner != "" {
			id = append(id, t.Owner...)
			id = append(id, '/')
		}
		id = append(id, t.Name...)
		if t.Revision != 0 {
			id = append(id, '/')
			id = strconv.AppendInt(id, int64(t.Revision), 10)
		}
		return string(id)
	}